cli/src/action/api.rs

```diff
@@ -19,17 +19,52 @@ use serde::Deserialize;
 
 use super::CliError;
 
-/// A wrapper around the Splinter REST API.
-pub struct SplinterRestClient<'a> {
-    pub url: &'a str,
+#[derive(Default)]
+pub struct SplinterRestClientBuilder {
+    pub url: Option<String>,
+    #[cfg(feature = "splinter-cli-jwt")]
+    pub jwt_auth: Option<String>,
 }
 
-impl<'a> SplinterRestClient<'a> {
-    /// Constructs a new client for a Splinter node at the given URL.
-    pub fn new(url: &'a str) -> Self {
-        Self { url }
+impl SplinterRestClientBuilder {
+    pub fn new() -> Self {
+        SplinterRestClientBuilder::default()
+    }
+
+    pub fn with_url(mut self, url: String) -> Self {
+        self.url = Some(url);
+        self
     }
 
+    #[cfg(feature = "splinter-cli-jwt")]
+    pub fn with_jwt_auth(mut self, jwt_auth: String) -> Self {
```

Comment:
This should just be `with_auth`, since the auth string isn't explicitly restricted to JWTs.

Reply:
Fixed